core/client/client.go

```diff
@@ -295,7 +295,7 @@ func (c *Client) CallAction(ctx context.Context, dbid string, action string, inp
 }
 
 // Call calls a procedure or action. It returns the result records.
-func (c *Client) Call(ctx context.Context, dbid string, procedure string, inputs []any) (*clientType.CallResult, error) {
+func (c *Client) Call(ctx context.Context, dbid string, procedure string, inputs []any, challenge []byte) (*clientType.CallResult, error) {
```

Comment:
Just a thought, but I don't see a compelling reason to expose the challenge or even knowledge of the authentication mechanism to the consumer of `Client`.  i.e. couldn't this be `Call(..., signReq bool)` and the method would do the challenge request internally?  Just for sake of discussion...  What is gained from explorting a `Challenge()` method?

Reply:
Ah, so one step further: make `signReq bool` a client config.  No new `Challenge` method and no change to `Call` or `Query`?  I'm inclined to hide away the challenge details.